src/app/services/git.service.ts

```diff
@@ -4,57 +4,47 @@ import * as os from 'os';
 import * as fs from 'fs';
 import * as child_process from 'child_process';
 import * as path from 'path';
-let shx = require('shelljs');
+import { which } from 'shelljs';
 
 export type TGitStatus = '?'|'*'|'\u2713';
 
 @Injectable()
 export class GITService {
-  currentBranch: string;
-  currentStatus: TGitStatus;
+  currentBranch: Promise<string>;
+  currentStatus: Promise<TGitStatus>;
 
   cwd: string;
   git: GITHandler;
 
   constructor(@Inject(ConfigService) private _config: ConfigService) {
     this.git = new GITHandler();
-    this.currentBranch = null;
-    this.currentStatus = '?';
     this.cwd = null;
   }
 
-  hasGit(): boolean {
-    if (!this.cwd) { return false; }
-    this.cwd = this.cwd.replace(/^~/, os.homedir());
-
-    try {
-      let branch: IGitResult = this.git.from(this.cwd).getBranch();
-      this.branch = !branch.code ? branch.value : null;
-
-      let status = this.git.from(this.cwd).getStatus();
-      this.status = !status.code ? this.parseStatus(status.value) : '?';
-      return true;
-    } catch (e) {
-      this.currentStatus = null;
-      this.currentBranch = null;
-      return false;
-    }
-  }
-
   parseStatus(s: string): TGitStatus {
+    if (s === null) { return '?'; }
```

Comment:
The reason for this:

> let s = ""
undefined
> s
''
>
> let a = null
undefined
> a
null
>
> !s
true
> !a
true
> s === a
false